### DISCUSSION TITLE
Conservative mode

### DISCUSSION THREAD

**Discussion by narwold** [renovatebot/renovate#16128] [Suggest an Idea]
We've been using Dependabot for Ruby updates to date, and it is more conservative when it comes to updating the Gemfile.lock. We would prefer this to avoid accidental breaking changes being swept up in a patch/minor update. For example, we have a PR for factory_bot v6.2.1 that does the following...

<img width="459" alt="Screen Shot 2022-06-17 at 9 36 14 AM" src="https://user-images.githubusercontent.com/1108152/174309224-6adc41a5-7c7f-4c48-8fa1-fc488a61e025.png">

...even though factory_bot's requirement for activesupport hasn't changed from `>= 5.0.0`.

My proposal is to add a `conservative: boolean` config option that allows the `--conservative` argument to be added to `bundler lock update` (see https://bundler.io/man/bundle-lock.1.html).

I have mocked out a PR for this already: https://github.com/renovatebot/renovate/pull/16129

**Comment by rarkins** [comment]
I would prefer to add this to post update options and definitely not a generic sounding top level config name like "conservative" which could confuse many users. The question is whether this should be default behaviour to opt out of instead of something to opt into?

**Reply by narwold** [reply]
"default behavior" - that was my thought, but I didn't know if it would be a major change for some folks that may have already been using it this way.

**Reply by narwold** [reply]
I also had a similar thought on "top level" being weird but didn't know where to put this. I opened a PR merely to dive in and get familiar with your codebase. I can update it as appropriate.

**Reply by narwold** [reply]
`postUpdateOptions` implies "after package/artifact updating" according to the docs, whereas this affects the behavior of the artifact updating itself. Would it still be OK there, or any other thoughts?

**Reply by rarkins** [reply]
I think we might end up renaming this one, including to reduce confusion with `postUpgradeTasks`, so it is ok to add this new feature there. Essentially it's now "artifact update options for package managers".

Regarding whether to make it default, the docs say "do not allow shared dependencies to be updated". That kind of implies that some upgrades might fail which today work? In which case we'd definitely want to start with it being opt-in instead of opt-out.
